internals/README-internals.md

```diff
@@ -40,3 +40,31 @@ See also DHCP propagation: https://github.com/coreos/fedora-coreos-config/pull/4
 
 See [this pull request](https://github.com/coreos/coreos-assembler/pull/768/commits/2701e91838e18d3eac0694fd0a5f003befcfb218) which added `/sysroot/.coreos-aleph-version.json` that can be used to track the version of that data.
 
+# ignition.platform.id
+
+The design we have today is that each CoreOS system is the same OS content - the same OSTree commit,
+and beyond that the exact same bootloader version, etc.
+
+There are differences per platform on the image formats (VHD versus qcow2 vs raw, etc).  However,
+what's *inside* the disk image for each platform is almost the same.
+
+A key difference between each image is the `ignition.platform.id` kernel argument.  From the
+moment the system boots and the kernel loads the initramfs, our userspace code uses this
+to reliably know its target platform.  As could be guessed from the name, [https://github.com/coreos/ignition/](ignition)
+uses this, and it runs early on.
+
+But there's other code which dynamically dispatches on the platform ID:
+
+- https://github.com/coreos/afterburn/
+- [The time sync setup code](https://github.com/coreos/fedora-coreos-config/blob/d87b52bc6a90b53e1afeab2731b52612d5e3bbc0/tests/kola/chrony/coreos-platform-chrony-generator#L9)
+- [network requirement detection](https://github.com/coreos/fedora-coreos-config/blob/d87b52bc6a90b53e1afeab2731b52612d5e3bbc0/overlay.d/05core/usr/lib/dracut/modules.d/35coreos-network/coreos-enable-network.service#L13)
+
+Notice in particular how the time synchronization code ends up reconfiguring chrony dynamically.
+For other operating systems which do "per cloud" disk images, it would have been more
+natural to just change `/etc/chrony.conf` per platform.  But that would mean we have a different
+ostree commit checksum per platform, breaking our "image based" update model.
+
+It's very unlikely that we will change the platform IDs in the future.  However, FCOS users are recommended
+to avoid parsing `ignition.platform.id`.  Generally, higher level code that needs to be
+platform aware will have more platform-specific ways to find this information.
```

Comment:
This seems a bit strong.  It's legitimate to have `ConditionKernelCommandLine=ignition.platform.id=x` in user-provided units.

Reply:
I hesitated on this for sure but it seemed easier to say no than yes :smile:  If we're allowing it we should document that...somewhere?

Reply:
https://docs.fedoraproject.org/en-US/fedora-coreos/platforms/